tests/lib/nsxt_unittest.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2015 The Capirca Project Authors All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""UnitTest class for nsxt.py."""
+
+import optparse
+from absl.testing import absltest
+from xml.etree import ElementTree as ET
+
+from capirca.lib import naming
+from capirca.lib import nsxt
+from capirca.lib import policy
+from capirca.tests.lib import nsxt_mocktest
+
+
+class TermTest(absltest.TestCase):
+
+  def setUp(self):
+    """Call before every test case."""
+    super().setUp()
+    parser = optparse.OptionParser()
+    parser.add_option(
+        '-d',
+        '--def',
+        dest='definitions',
+        help='definitions directory',
+        default='../def')
+    (FLAGS, args) = parser.parse_args()
+    self.defs = naming.Naming(FLAGS.definitions)
+
+  def tearDown(self):
+    super().setUp()
+    pass
+
+  def runTest(self):
+    pass
+
+  def test_init_forinet(self):
+    """Test for Term._init_."""
+    inet_term = nsxt.Term(nsxt_mocktest.INET_TERM, 'inet')
+    self.assertEqual(inet_term.af, 4)
+    self.assertEqual(inet_term.filter_type, 'inet')
+
+  def test_init_forinet6(self):
+    """Test for Term._init_."""
+    inet6_term = nsxt.Term(nsxt_mocktest.INET6_TERM, 'inet6', 6)
+    self.assertEqual(inet6_term.af, 6)
+    self.assertEqual(inet6_term.filter_type, 'inet6')
+
+  def test_ServiceToStr(self):
+    """Test for Term._ServiceToStr."""
+
+    proto = 6
+    icmp_types = []
+    dports = [(1024, 65535)]
+    spots = [(123, 123)]
+    nsxt_term = nsxt.Term(nsxt_mocktest.INET_TERM, 'inet')
+    service = nsxt_term._ServiceToString(proto, spots, dports, icmp_types)
+    self.assertEqual(
+        service, '<service><protocol>6</protocol><sourcePort>123'
```

Comment:
The expected generated service term should be JSON, not XML, right?

Reply:
All the XML content was removed with the last commit, it was a bit of a mess moving from one repo to another.